src/fragments/lib/auth/js/mfa.mdx

```diff
@@ -14,81 +14,184 @@ With TOTP (Time-based One-time Password), your app user is challenged to complet
-});
 
-// ...
+  // ...
 
-// Then you will have your TOTP account in your TOTP-generating app (like Google Authenticator)
-// Use the generated one-time password to verify the setup
-Auth.verifyTotpToken(user, challengeAnswer)
-  .then(() => {
+  // Then you will have your TOTP account in your TOTP-generating app (like Google Authenticator)
+  // Use the generated one-time password to verify the setup
+  try {
+    const cognitoUserSession: CognitoUserSession = await Auth.verifyTotpToken(
+      user,
+      challengeAnswer
+    );
     // don't forget to set TOTP as the preferred MFA method
-    Auth.setPreferredMFA(user, 'TOTP');
-    // ...
-  })
-  .catch((e) => {
+    await Auth.setPreferredMFA(user, 'TOTP');
+  } catch (error) {
     // Token is not verified
-  });
+  }
+
+  // ...
+
+  // Finally, when sign-in with MFA is enabled, use the confirmSignIn method
+  // to pass the TOTP code and MFA type.
+  await Auth.confirmSignIn(user, tOTPCode, mfaType); // Optional, MFA Type e.g. SMS_MFA || SOFTWARE_TOKEN_MFA
+}
+```
+</Block>
+<Block name="JavaScript">
+
+```js
+import { Auth } from 'aws-amplify';
+
+async function setupTOTPAuth(user, challengeAnswer, mfaType) {
+  // To setup TOTP, first you need to get a `authorization code` from Amazon Cognito
+  // `user` is the current Authenticated user
+  const tOTPCode = await Auth.setupTOTP(user);
+  // You can directly display the `code` to the user or convert it to a QR code to be scanned.
+  // E.g., use following code sample to render a QR code with `qrcode.react` component:
+  //      import QRCodeCanvas from 'qrcode.react';
+  //      const str = "otpauth://totp/AWSCognito:"+ username + "?secret=" + code + "&issuer=" + issuer;
+  //      <QRCodeCanvas value={str}/>
 
-// ...
+  // ...
 
-// Finally, when sign-in with MFA is enabled, use the confirmSignIn method 
-// to pass the TOTP code and MFA type. 
-Auth.confirmSignIn(user, code, mfaType) // Optional, MFA Type e.g. SMS_MFA || SOFTWARE_TOKEN_MFA
+  // Then you will have your TOTP account in your TOTP-generating app (like Google Authenticator)
+  // Use the generated one-time password to verify the setup
+  try {
+    const cognitoUserSession = await Auth.verifyTotpToken(user, challengeAnswer);
+    // don't forget to set TOTP as the preferred MFA method
+    await Auth.setPreferredMFA(user, 'TOTP');
+  } catch (error) {
+    // Token is not verified
+  }
 
+  // ...
 
+  // Finally, when sign-in with MFA is enabled, use the confirmSignIn method
+  // to pass the TOTP code and MFA type.
+  await Auth.confirmSignIn(user, code, mfaType); // Optional, MFA Type e.g. SMS_MFA || SOFTWARE_TOKEN_MFA
+}
 ```
+</Block>
+</BlockSwitcher>
 
 ## Setup MFA type
 
 Multiple MFA types supported by Amazon Cognito. You can set the preferred method in your code:
 
-```javascript
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+import { Auth } from 'aws-amplify';
+
+type SetupMFATypeParameters = {
+  user: string;
+};
```

Comment:
We can remove this code as user should come from `Auth.currentAuthenticatedUser` as explained bellow.